utils/mysql/files/mysqld.init

```diff
@@ -7,24 +7,31 @@ STOP=10
 SERVICE_DAEMONIZE=1
 SERVICE_WRITE_PID=1
 SERVICE_STOP_TIME=9
-
-error() {
-	echo "${initscript}:" "$@" 1>&2
-}
+PROG='/usr/bin/mysqld'
 
 start() {
-	local datadir=$(sed -n -e "s/^[[:space:]]*datadir[[:space:]]*=[[:space:]\"']*\([^[:space:]\"']*\)[[:space:]\"']*/\1/p" /etc/my.cnf)
-	if [ ! -d "$datadir" ]; then
-		error "Error: datadir '$datadir' in /etc/my.cnf doesn't exist"
+	local datadir
+	local config='/etc/my.cnf'
+
+	# e.g.: datadir  = '/my/path/with spaces'
+	set -- $( grep ^'datadir' "$config" )
+	shift 2
+	eval datadir="$*"
```

Comment:
This breaks unquoted paths with spaces:

    # (set -x; set -- $( echo 'datadir = foo bar' ); shift 2; eval datadir="$*")
    ++ echo 'datadir = foo bar'
    + set -- datadir = foo bar
    + shift 2
    + eval 'datadir=foo bar'
    ++ datadir=foo
    ++ bar
    bash: bar: command not found

It is also still prone to shell injections:

    # (set -x; set -- $( echo 'datadir = $(date)' ); shift 2; eval datadir="$*")
    ++ echo 'datadir = $(date)'
    + set -- datadir = '$(date)'
    + shift 2
    + eval 'datadir=$(date)'
    +++ date
    ++ datadir='Do 1. Dez 10:10:27 CET 2016'

Whats wrong with the propsed sed command? It is safer, requires less forks, does not rely on eval and handles all corner cases nicely.